### PR TITLE
Editable identity on profile page

### PR DESCRIPTION
**Pull request by hgwood**
This is part of #36.

### CODE REVIEW

**Comment by hgwood** [comment]
You might want to rebase this instead of squashing so that the React update stand out. I can also rollback the react update if you want.

I updated React because I wanted to use the new `getDerivedStateFromProps` new API, but it turns out I cannot because Apollo React rely on the "old" (pre-16.3) lifecycle API and React forbids to use both APIs at the same time.

If you want to rollback the update, you need to remove the `UNSAFE_` prefix on `componentWillReceiveProps` on `UserProfile`.

**Comment by Errorname** [comment]
I just created a PR #62 which upgrade React to 16.3. It also updates react-apollo to make getDerivedStateFromProps works in everycase.

**Comment by hgwood** [comment]
OK I'll update this PR to match.

**Comment by hgwood** [comment]
Done.

**Comment by hgwood** [comment]
Well, no, not quite done actually.

**Comment by Errorname** [comment]
I added a small ui feedback when saving and the Avatar component to see the profile picture

File: src/scenes/UserProfile/UserProfile.jsx [hunk 1]
```diff
@@ -90,19 +91,29 @@ class UserProfile extends Component {
...
               <label htmlFor="name">Name</label>
-              <input name="name" className="card-input" value={name} onChange={e => this.onIdentityChange(e)} />
+              <input name="name" className="card-input" value={name} onChange={e => this.onIdentityChange(e)} autoComplete="off" />
               <label htmlFor="email">Email address</label>
               <input name="email" className="card-input" value={email} onChange={e => this.onIdentityChange(e)} />
               <label htmlFor="picture">Picture link</label>
-              <input name="picture" className="card-input" value={picture} onChange={e => this.onIdentityChange(e)} />
-            </form>
+							<div style={{display: 'flex', alignItems: 'center'}}>
+								<Avatar
```
Comment (hgwood) [inline src/scenes/UserProfile/UserProfile.jsx]: Indentation looks weird. Maybe you used tabs?

File: src/scenes/UserProfile/UserProfile.jsx [hunk 2]
```diff
@@ -90,19 +91,29 @@ class UserProfile extends Component {
...
+							</div>
+						</form>
             <CardActions>
               <Button
                 primary
                 type="button"
                 disabled={savingIdentity}
                 onClick={() => this.updateIdentity(User.id, this.state.identity)}>
-                Save
+                {savingIdentity?'Saving...':'Save'}
```
Comment (hgwood) [inline src/scenes/UserProfile/UserProfile.jsx]: Indeed! But I think you forgot to ask Prettier to format this line :)

File: src/scenes/UserProfile/UserProfile.jsx [hunk 3]
```diff
@@ -90,19 +91,29 @@ class UserProfile extends Component {
...
-            </form>
+							<div style={{display: 'flex', alignItems: 'center'}}>
+								<Avatar
+                  image={User.picture}
+                  style={{
+                    width: '60px',
+                    height: '60px',
+                    marginRight: '20px'
+                  }}
+                />
```
Comment (hgwood) [inline src/scenes/UserProfile/UserProfile.jsx]: Could you show a screenshot of what it looks like?

Comment (Errorname) [reply src/scenes/UserProfile/UserProfile.jsx]: image

**Comment by hgwood** [comment]
Thanks. Looks good to me. You may merge when ready.
image

**Pull request merged**